File: app/src/main/java/com/abaffym/moviesapp/ui/favorites/FavoritesViewModel.java
package com.abaffym.moviesapp.ui.favorites;

import com.abaffym.moviesapp.model.Movie;
import com.abaffym.moviesapp.ui.MoviesRepository;

import java.util.List;

import androidx.lifecycle.ViewModel;
import io.reactivex.Single;

public class FavoritesViewModel extends ViewModel {

	private MoviesRepository movieRepository;

	FavoritesViewModel() {
		this.movieRepository = new MoviesRepository();
	}

	public Single<List<Movie>> getFavoriteMovies() {
		return movieRepository.getFavoriteMovies();
	}

}

File: app/src/main/java/com/abaffym/moviesapp/ui/favorites/FavoritesViewModel.kt
package com.abaffym.moviesapp.ui.favorites

import android.arch.lifecycle.ViewModel
import com.abaffym.moviesapp.ui.MoviesRepository

class FavoritesViewModel : ViewModel() {

    private val movieRepository: MoviesRepository = MoviesRepository()

    val favoriteMovies = movieRepository.favoriteMovies

}
